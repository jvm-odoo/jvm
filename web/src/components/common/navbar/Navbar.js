import React from 'react'
import Fade from 'react-reveal/Fade'

import {
  NavbarContainer,
  NavbarMenu,
  NavbarLogo,
  NavbarButton
} from './'

import { Container } from '../'

export const Navbar = () => (
  <NavbarContainer>
    <Container flex fullHeight>
      <Fade top>
        <NavbarLogo to='/'>
          <span>J</span>
          <span>V</span>
          <span>M</span>
        </NavbarLogo>
      </Fade>
      <NavbarMenu>
        {/* <Fade top>
          <NavbarMenuItem>
            <Link to="/">About</Link>
          </NavbarMenuItem>
          <NavbarMenuItem>
            <Link to="/">Lab</Link>
          </NavbarMenuItem>
          <NavbarMenuItem>
            <Link to="/">Blog</Link>
          </NavbarMenuItem>
        </Fade> */}
      </NavbarMenu>
      <Fade top>
        <NavbarButton to="/">
          Contact
        </NavbarButton>
      </Fade>
    </Container>
  </NavbarContainer>
)
