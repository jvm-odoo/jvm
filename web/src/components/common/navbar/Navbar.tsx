import React from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { useHistory, Link } from 'react-router-dom'

import {
    NavbarContainer,
    NavbarMenu,
    NavbarMenuItem,
    NavbarLogo,
    NavbarButton
} from './'

import { Container } from '../'
import {ISession, setToken} from '../../../redux/slices/session'

interface INavbarProps {
    contact?: boolean,
    dashboard?: boolean
}

export const Navbar = ({ contact, dashboard }: INavbarProps) => {
    const history = useHistory()
    const session = useSelector(({ session }: ISession) => session)
    const dispatch = useDispatch()

    const handleContactButtonClick = (e) => {
        e.preventDefault()
        window.location.href = 'mailto:jasonvanmalder@gmail.com'
    }

    const handleSignOutButtonClick = (e) => {
        e.preventDefault()
        dispatch(setToken(undefined))
        history.push('/')
    }

    return (
        <NavbarContainer>
            <Container flex fullHeight>
                <NavbarLogo to='/'>
                    <span>J</span>
                    <span>V</span>
                    <span>M</span>
                </NavbarLogo>
                <NavbarMenu>
                    {/*
            <NavbarMenuItem>
              <Link to="/">About</Link>
            </NavbarMenuItem>
            <NavbarMenuItem>
              <Link to="/">Lab</Link>
            </NavbarMenuItem>
            <NavbarMenuItem>
              <Link to="/">Blog</Link>
            </NavbarMenuItem>
          */}
                    { dashboard && (
                        <NavbarMenuItem>
                            <Link to="/dashboard/">Landing Data</Link>
                        </NavbarMenuItem>
                    )}
                </NavbarMenu>
                { contact && (
                    <NavbarButton aria-label='Contact' to="/" onClick={handleContactButtonClick}>
                        Contact
                    </NavbarButton>
                )}
                { !contact && session.token && (
                    <NavbarButton aria-label='Sign out' to="/" onClick={handleSignOutButtonClick}>
                        Sign out
                    </NavbarButton>
                )}
            </Container>
        </NavbarContainer>
    )
}