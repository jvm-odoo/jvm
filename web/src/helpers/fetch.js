export const fetchGet = async (url) => {
  const response = await fetch(url)
  return response.json()
}
