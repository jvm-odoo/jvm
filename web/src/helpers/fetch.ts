export const fetchGet = async (url: string) => {
    const response = await fetch(url)
    return response.json()
}

export const fetchPost = async (url: string, data: any, token?: string) => {
    const headers: Headers = new Headers({
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*'
    })

    if (token) {
        headers.append('Authorization', `Token ${token}`)
    }

    const response = await fetch(url, {
        method: 'POST',
        headers,
        mode: 'cors',
        body: JSON.stringify(data)
    })

    return response.json()
}