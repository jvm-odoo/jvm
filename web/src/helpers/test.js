import React from 'react'
import toJson from 'enzyme-to-json'
import { act } from 'react-dom/test-utils'
import { mount, shallow } from 'enzyme'
import { MockedProvider } from '@apollo/react-testing'

const wait = (time = 0) => new Promise(res => setTimeout(res, time))

export const executeMockProviderTestCase = async (wrapperInstance) => (
  await act(async () => {
    await wait(100)
    wrapperInstance.update()
  })
)

export const shallowMatchSnapshot = (Component) => {
  const wrapper = shallow(Component)
  expect(toJson(wrapper)).toMatchSnapshot()
}

export const mountWrapperWithMockedData = async (Component, mocks) => {
  let wrapper

  act(() => {
    wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        {Component}
      </MockedProvider>
    )
  })

  await executeMockProviderTestCase(wrapper)
  return wrapper
}
