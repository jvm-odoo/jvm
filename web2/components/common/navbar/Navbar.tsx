import React, { MouseEvent, useRef, useState } from 'react'
import { useSelector, useDispatch } from 'react-redux'

import { NavbarButton, NavbarContainer, NavbarLogo } from './'
import { ISessionReducer, setToken } from '../../../redux/slices/session'
import { AppBar, Button, Toolbar, Typography } from '@material-ui/core'
import styled from 'styled-components'
import { NavbarButtons } from './NavbarButton'

interface INavbarProps {
    contact?: boolean
    dashboard?: boolean
    signin?: boolean
}

const NavbarAppBar = styled(AppBar)`
    background-color: transparent;
    box-shadow: none;
`

const NavbarToolbar = styled(Toolbar)`
    justify-content: space-between;
    padding: 0;
    height: 80px;
`

export const Navbar = ({ contact, dashboard, signin }: INavbarProps) => {
    const session = useSelector(({ session }: ISessionReducer) => session)
    const dispatch = useDispatch()
    const navbarMenu = useRef(null)
    const [isMenuOpen, setIsMenuOpen] = useState(false)

    const handleContactButtonClick = (e: MouseEvent) => {
        e.preventDefault()
        window.location.href = 'mailto:jasonvanmalder@gmail.com'
    }

    const handleSignOutButtonClick = (e: MouseEvent) => {
        e.preventDefault()
        dispatch(setToken(undefined))
        // history.push('/')
    }

    const toggleMobileMenu = () => {
        setIsMenuOpen((currentIsMenuOpen: boolean) => {
            const menu = navbarMenu.current as HTMLElement | null

            if (!currentIsMenuOpen) {
                menu?.classList?.add('show')
                document.body.classList.add('stop-scrolling')
            } else {
                menu?.classList?.remove('show')
                document.body.classList.remove('stop-scrolling')
            }

            return !currentIsMenuOpen
        })
    }

    const shouldShowContact = !localStorage.getItem('jvnm')
    const shouldShowSignIn = !shouldShowContact && !signin

    const getLogo = () => {
        if (!dashboard) {
            return (
                <>
                    <span>Jason</span>
                    <span>&nbsp;Van Malder</span>
                </>
            )
        }

        const firstPart = 'To'
        const secondPart = 'Do'

        return (
            <>
                <span>{firstPart}</span>
                <span>{secondPart}</span>
            </>
        )
    }

    return (
        <NavbarContainer>
            <NavbarAppBar position="static">
                <NavbarToolbar>
                    <NavbarLogo to="/">{getLogo()}</NavbarLogo>

                    <NavbarButtons>
                        {contact && shouldShowContact && (
                            <NavbarButton
                                aria-label="Contact"
                                to="/"
                                onClick={handleContactButtonClick}
                            >
                                Contact
                            </NavbarButton>
                        )}

                        {shouldShowSignIn && !session.token && (
                            <NavbarButton aria-label="Sign in" to="/signin">
                                Sign in
                            </NavbarButton>
                        )}

                        {shouldShowSignIn && session.token && !dashboard && (
                            <NavbarButton
                                aria-label="dashboard"
                                to="/dashboard"
                            >
                                Dashboard
                            </NavbarButton>
                        )}

                        {!contact && session.token && (
                            <NavbarButton
                                aria-label="Sign out"
                                to="/"
                                onClick={handleSignOutButtonClick}
                            >
                                Sign out
                            </NavbarButton>
                        )}
                    </NavbarButtons>
                </NavbarToolbar>
            </NavbarAppBar>
        </NavbarContainer>
    )
}
