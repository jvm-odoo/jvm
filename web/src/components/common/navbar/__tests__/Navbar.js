import React from 'react'

import { Navbar } from '../Navbar'
import { shallowMatchSnapshot } from '../../../../helpers/test'

it('renders correctly', () => shallowMatchSnapshot(<Navbar />))
