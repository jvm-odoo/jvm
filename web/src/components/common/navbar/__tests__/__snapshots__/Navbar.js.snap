// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.nav>
  <styled.div
    flex={true}
    fullHeight={true}
  >
    <Fade
      top={true}
    >
      <Styled(Link)
        to="/"
      >
        <span>
          J
        </span>
        <span>
          V
        </span>
        <span>
          M
        </span>
      </Styled(Link)>
    </Fade>
    <styled.ul />
    <Fade
      top={true}
    >
      <Styled(Styled(Link))
        onClick={[Function]}
        to="/"
      >
        Contact
      </Styled(Styled(Link))>
    </Fade>
  </styled.div>
</styled.nav>
`;
